.github/workflows/pull_request_soundness.yml

```diff
@@ -125,10 +125,10 @@ jobs:
       uses: actions/checkout@v4
     - name: Mark the workspace as safe
     # https://github.com/actions/checkout/issues/766
-      run: git config --global --add safe.directory ${GITHUB_WORKSPACE}
+      run: if [ -n "${GITHUB_WORKSPACE:-}" ]; then git config --global --add safe.directory ${GITHUB_WORKSPACE}; fi
```

Comment:
Can we put the if check into a separate job `if:` condition instead of into the script

Reply:
`if:` is only on the `job` not the `steps`. Are you hung up on this one, if so I'll just remove this and update my `yq` hack to filter it out (which I'd rather not, but don't want to block progress).

Reply:
That's not correct. `steps` support `if` conditions https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstepsif

Reply:
I am happy to add this but can we also add it to the API breaking job that does the same

Reply:
You were right about `steps[*]` supporting `if`, sorry about that.

Done, and updated the API check job to do the same, as requested.

Reply:
Hm, I used the exact syntax from the example docs: https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#example-using-secrets.

```yaml
    steps:
      - if: ${{ env.super_secret != '' }}
        run: echo 'This step will only run if the secret has a value set.'
      - if: ${{ env.super_secret == '' }}
        run: echo 'This step will only run if the secret does not have a value set.'
```

But it's failing.

Reply:
OK, I ditched this and also fixed the diff command while there to exit non-zero with changes and actually show what they are.